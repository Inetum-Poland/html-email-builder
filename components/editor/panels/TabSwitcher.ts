import type { PanelProperties } from "grapesjs";
import { Paintbrush, Settings, Layers, Blocks } from "lucide-static";

export const TabSwitcher = (): PanelProperties => {
  const { $i18n: { t } } = useNuxtApp();

  return {
    id: "tab-switcher",
    buttons: [
      {
        id: "style-component",
        label: Paintbrush,
        command: "open-sm",
        active: true,
        togglable: false,
        attributes: { id: "style-component", title: t("styleComponent") },
        tagName: "button",
      },
      {
        id: "edit-component-settings",
        label: Settings,
        command: "open-tm",
        togglable: false,
        attributes: {
          id: "edit-component-settings",
          title: t("editComponentSettings"),
        },
        tagName: "button",
      },
      {
        id: "view-layers",
        label: Layers,
        command: "open-layers",
        togglable: false,
        attributes: { id: "view-layers", title: t("viewLayers") },
        tagName: "button",
      },
      {
        id: "add-blocks",
        label: Blocks,
        command: "open-blocks",
        togglable: false,
        attributes: { id: "add-blocks", title: t("addBlocks") },
        tagName: "button",
      },
    ],
  };
};
