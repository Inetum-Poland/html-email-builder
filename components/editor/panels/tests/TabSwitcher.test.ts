import { expect, test } from "vitest";
import { TabSwitcher } from "../TabSwitcher";

test("renders the Tab Switcher panel", () => {
  expect(TabSwitcher()).toMatchSnapshot();
});
