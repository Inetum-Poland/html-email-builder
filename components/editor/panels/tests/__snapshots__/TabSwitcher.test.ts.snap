// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders the Tab Switcher panel 1`] = `
{
  "buttons": [
    {
      "active": true,
      "attributes": {
        "id": "style-component",
        "title": "Style component",
      },
      "command": "open-sm",
      "id": "style-component",
      "label": "
<svg
  class="lucide lucide-paintbrush"
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <path d="m14.622 17.897-10.68-2.913" />
  <path d="M18.376 2.622a1 1 0 1 1 3.002 3.002L17.36 9.643a.5.5 0 0 0 0 .707l.944.944a2.41 2.41 0 0 1 0 3.408l-.944.944a.5.5 0 0 1-.707 0L8.354 7.348a.5.5 0 0 1 0-.707l.944-.944a2.41 2.41 0 0 1 3.408 0l.944.944a.5.5 0 0 0 .707 0z" />
  <path d="M9 8c-1.804 2.71-3.97 3.46-6.583 3.948a.507.507 0 0 0-.302.819l7.32 8.883a1 1 0 0 0 1.185.204C12.735 20.405 16 16.792 16 15" />
</svg>
",
      "tagName": "button",
      "togglable": false,
    },
    {
      "attributes": {
        "id": "edit-component-settings",
        "title": "Edit component settings",
      },
      "command": "open-tm",
      "id": "edit-component-settings",
      "label": "
<svg
  class="lucide lucide-settings"
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <path d="M12.22 2h-.44a2 2 0 0 0-2 2v.18a2 2 0 0 1-1 1.73l-.43.25a2 2 0 0 1-2 0l-.15-.08a2 2 0 0 0-2.73.73l-.22.38a2 2 0 0 0 .73 2.73l.15.1a2 2 0 0 1 1 1.72v.51a2 2 0 0 1-1 1.74l-.15.09a2 2 0 0 0-.73 2.73l.22.38a2 2 0 0 0 2.73.73l.15-.08a2 2 0 0 1 2 0l.43.25a2 2 0 0 1 1 1.73V20a2 2 0 0 0 2 2h.44a2 2 0 0 0 2-2v-.18a2 2 0 0 1 1-1.73l.43-.25a2 2 0 0 1 2 0l.15.08a2 2 0 0 0 2.73-.73l.22-.39a2 2 0 0 0-.73-2.73l-.15-.08a2 2 0 0 1-1-1.74v-.5a2 2 0 0 1 1-1.74l.15-.09a2 2 0 0 0 .73-2.73l-.22-.38a2 2 0 0 0-2.73-.73l-.15.08a2 2 0 0 1-2 0l-.43-.25a2 2 0 0 1-1-1.73V4a2 2 0 0 0-2-2z" />
  <circle cx="12" cy="12" r="3" />
</svg>
",
      "tagName": "button",
      "togglable": false,
    },
    {
      "attributes": {
        "title": "View layers",
      },
      "command": "open-layers",
      "id": "view-layers",
      "label": "
<svg
  class="lucide lucide-layers"
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <path d="m12.83 2.18a2 2 0 0 0-1.66 0L2.6 6.08a1 1 0 0 0 0 1.83l8.58 3.91a2 2 0 0 0 1.66 0l8.58-3.9a1 1 0 0 0 0-1.83Z" />
  <path d="m22 17.65-9.17 4.16a2 2 0 0 1-1.66 0L2 17.65" />
  <path d="m22 12.65-9.17 4.16a2 2 0 0 1-1.66 0L2 12.65" />
</svg>
",
      "tagName": "button",
      "togglable": false,
    },
    {
      "attributes": {
        "title": "Add blocks",
      },
      "command": "open-blocks",
      "id": "add-blocks",
      "label": "
<svg
  class="lucide lucide-blocks"
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <rect width="7" height="7" x="14" y="3" rx="1" />
  <path d="M10 21V8a1 1 0 0 0-1-1H4a1 1 0 0 0-1 1v12a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1v-5a1 1 0 0 0-1-1H3" />
</svg>
",
      "tagName": "button",
      "togglable": false,
    },
  ],
  "id": "tab-switcher",
}
`;
